3rdparty/optee/libutee/entry.c

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include <compiler.h>
+#include <tee_ta_api.h>
+#include <utee_syscalls.h>
+
+struct utee_params;
+
+TEE_Result __utee_entry(
+    unsigned long func,
+    unsigned long session_id,
+    struct utee_params* up,
+    unsigned long cmd_id);
+
+void __ta_entry(
```

Comment:
why do you need 16-20 instead of just using 22-26?

Reply:
GCC specifies that attributes are to be attached to function declarations.

I've done some experimentation removing the declaration and leaving only the definition:
1. If the attribute is placed at the end of a function definition, GCC emits an error;
2. If the attribute is placed at the beginning of a function definition, GCC does not complain.

However, all documentation points to attributes being expected to be placed on function declarations and not definitions.